proposals/0020-no-type-in-type.rst

```diff
@@ -285,7 +277,58 @@ Alternatives
    I find both arguments compelling independently, and so I withdraw support for this
    alternative. Nevertheless, I'm keeping it in the proposal in case someone wants to
    argue in support of it.
-   
+
+Implication TypeOperators to NoStarIsType
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One alternative is to have an additional 4.f step in the Proposed Change Specification section:
+
+   f. For two releases, ``-XTypeOperators`` will imply ``-XNoStarIsType``, to
+      provide a migration path for code that uses the binary operator ``*``. (After
+      two releases, this code can include ``-XNoStarIsType`` explicitly without
+      going against the three-release policy.) Users can re-enable ``-XStarIsType``
+      after ``-XTypeOperators`` is enabled if they wish.
+
+This alternative is problematic. It's intended to help migration, but
+implementation evidence shows it causes more trouble.  If the code being
+migrated only enables ``-XTypeOperators``, then 4.f will indeed be helpful, but
+there is a lot of code which also enables ``-XKindSignatures`` in which
+``-XNoStarInType`` changes the semantics. For example code from ``hashable`` library
+
+::
+
+    newtype  Tagged  (s :: * -> *) = Tagged {unTagged :: Int}
+
+fails to compiler with the following error:
+
+::
+
+  Data/Hashable/Generic.hs:116:22: error:
+      Operator applied to too few arguments: *
+      With NoStarIsType (implied by TypeOperators), ‘*’ is treated as a regular type operator.
+      Did you mean to use ‘Type’ from Data.Kind instead?
+      |
+  116 | newtype Tagged (s :: * -> *) = Tagged {unTagged :: Int}
+      |
+
+Many other libraries in the wild simultaneously enable ``-XTypeOperators`` and use
+``*`` as a kind, including servant, aeson, cereal, cassava, and others, so
+including 4.f would break all of these libraries. While not having 4.f also
+results in some breakage, far fewer libraries in the wild break without 4.f
+than with 4.f, so this was ultimately decided against.
+
+These modules will suddenly have ``-XNoStarIsType`` in effect, meaning that
+their use of ``*`` will refer to a binary operator. These modules have a
+choice of how to proceed. They can either:
```

Comment:
I would replace this whole paragraph with:

"If `TypeOperators` were to imply `NoStarIsType`, then any code which uses `*` as kind instead of a binary operator would have to migrate somehow. Two migration options are:"